components/browser/state/src/main/java/mozilla/components/browser/state/reducer/LocaleStateReducer.kt

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.reducer
+
+import mozilla.components.browser.state.action.LocaleAction
+import mozilla.components.browser.state.state.BrowserState
+
+internal object LocaleStateReducer {
+
+    /**
+     * [LocaleAction] Reducer function for modifying [BrowserState.locale].
+     */
+    fun reduce(state: BrowserState, action: LocaleAction): BrowserState {
+        return when (action) {
+            is LocaleAction.UpdateLocaleAction -> state.copy(locale = action.locale)
+            is LocaleAction.RestoreLocaleStateAction -> updateLocaleState(state)
```

Comment:
See https://github.com/mozilla-mobile/android-components/pull/9779/files#r592740567 here you just have to return is `LocaleAction.RestoreLocaleStateAction -> state`

Reply:
That seemed too easy 😆